service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/RestoreAzureDatabaseStep.java

```diff
@@ -0,0 +1,126 @@
+
+    var inputParameters = context.getInputParameters();
+    var workingMap = context.getWorkingMap();
+
+    ControlledAzureDatabaseResource destinationDatabase =
+        getRequired(
+            workingMap,
+            WorkspaceFlightMapKeys.ControlledResourceKeys.CLONED_RESOURCE_DEFINITION,
+            ControlledAzureDatabaseResource.class);
+
+    var destinationWorkspaceId =
+        getRequired(
+            inputParameters,
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID,
+            UUID.class);
+    var userRequest =
+        getRequired(
+            inputParameters,
+            JobMapKeys.AUTH_USER_INFO.getKeyName(),
+            AuthenticatedUserRequest.class);
+
+    var destinationContainer =
+        getRequired(
+            workingMap,
+            WorkspaceFlightMapKeys.ControlledResourceKeys.AZURE_STORAGE_CONTAINER,
+            ControlledAzureStorageContainerResource.class);
+
+    var sasToken =
+        azureStorageAccessService.createAzureStorageContainerSasToken(
+            destinationWorkspaceId, destinationContainer, userRequest, null, null, "rcwl");
+    var blobContainerUrlAuthenticated = sasToken.sasUrl();
+
+    var blobFileName =
+        workingMap.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.CLONE_DB_DUMPFILE, String.class);
+
+    // Query LZ for the postgres server info
+    var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
+    var landingZoneId =
+        landingZoneApiDispatch.getLandingZoneId(
+            bearerToken, workspaceService.getWorkspace(destinationDatabase.getWorkspaceId()));
+    var dbServerName =
+        getResourceName(
+            landingZoneApiDispatch
+                .getSharedDatabase(bearerToken, landingZoneId)
+                .orElseThrow(() -> new RuntimeException("No shared database found")));
+    var adminDbUserName =
+        getResourceName(
+            landingZoneApiDispatch
+                .getSharedDatabaseAdminIdentity(bearerToken, landingZoneId)
+                .orElseThrow(
+                    () -> new RuntimeException("No shared database admin identity found")));
+    var dumpEncryptionKey =
+        workingMap.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.CLONE_DB_DUMP_ENCRYPTION_KEY,
+            String.class);
+
+    this.azureDatabaseUtilsRunner.pgRestoreDatabase(
+        workingMap.get(AZURE_CLOUD_CONTEXT, AzureCloudContext.class),
+        destinationWorkspaceId,
+        "restore-db-" + destinationDatabase.getResourceId(),
+        destinationDatabase.getDatabaseName(),
+        dbServerName,
+        adminDbUserName,
+        blobFileName,
+        destinationContainer.getStorageContainerName(),
+        blobContainerUrlAuthenticated,
+        dumpEncryptionKey);
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    // Nothing to undo in this step
```

Comment:
update this comment to point to the place where the database is deleted (making it unnecessary to undo this "restore" step)